File: p4/p4.erl
#!/usr/bin/env escript
%% -*- erlang -*-
%%! -smp enable -sname p4
% vim:syn=erlang

-mode(compile).

main(_) ->
    Answer = lists:max([ X*Y || X <- lists:seq(999,100,-1), Y <- lists:seq(999,100,-1), is_palindrome(X*Y)]),
    io:format("Answer ~p ~n",[ Answer ]).


is_palindrome(N) -> S = integer_to_list(N), S =:= lists:reverse(S).

File: p4/p4.pl
#!/usr/bin/perl -w
use strict;

my $answer = 0;
for my $x (-999..-100){
    for my $y ($x..-100){
        my $n = $x*$y;
        $answer = $n if $n > $answer && $n == reverse(split(//,$n))   
    }    
}
print "Answer: $answer \n";
